### DISCUSSION TITLE
Changing what the empty string does for the playbgm controller

### DISCUSSION THREAD

**Discussion by Seb471** [ikemen-engine/Ikemen-GO#1694] [Feature Request]
### Is your feature request related to a problem? Please describe.

Not quite a problem, but basically there is no way to reset a bgm to be the original bgm of the stage music you changed from.

Basically I tried using playbgm to make an install music that would play when a character enters an install state.
Entering it isn't the problem, it's getting out of it which is the problem.

Right now, when using the playbgm controller, there are 2 ways we can use it :
- Change a music to be one that is within the files of the character by specifying it in the bgm path
- Change a music to be nothing if it's the bgm path isn't specified

The latter implementation seems a bit redundant, since the assert special controller with the flag no music does a similar job to this.

### Describe the solution you'd like

Changing the behaviour of what an empty bgm path does to the playbgm controller.
Instead of playing nothing it should instead replay the bgm of the stage in question so that we can "go back to neutral" kind of.

### Additional context

This feature has been discussed in #ikemen-help on discord with users Foobs and PlasmoidThunder which came up with the idea of this change feature and Super for proposing to put this feature request in here

### Search terms

playbgm stage bgm install controller music

**Comment by RealFoobs** [comment]
I would like this feature, just that my ideal implementation would be a new parameter like "stagegbm", over declaring an empty path. That's just inconsistent with how playsnd works.

**Reply by Seb471** [reply]
This is also a good alternative.
But I don't get the "inconsistent with how playsnd works" part...
Since I don't think you can put nothing as a value in playsnd, to have well, nothing.

**Comment by LasombraDemon** [comment]
A possible solution would be to be able to point to the Stage's Original BGM as StageVar(music.BGMname) or something. That's referring to data as a string, though...

**Comment by potsmugen** [comment]
Yeah I also think empty field should do nothing or stop the BGM (or a StopBGM controller for that one?).

Regarding the stage BGM, since we can now have more than one per stage (faster versions like SF2), I wonder if a stage BGM parameter should pick the correct one automatically or just play the default one.
